www-src/app/lib/compare_versions.coffee

```diff
@@ -0,0 +1,21 @@
+
+module.exports.splitVersion = splitVersion = (s) ->
+    unless s?
+        return major: 0, minor: 0, patch: 0
+
+    parts = s.match /(\d+)\.(\d+)\.(\d+)/
+    # Keep only useful data (first elem is full string)
+    parts = parts.slice 1, 4
+    parts = parts.map (s) -> parseInt s
+    return major: parts[0], minor: parts[1], patch: parts[2]
+
+
+module.exports.compareVersions = (version1, version2) ->
+    v1 = splitVersion version1
+    v2 = splitVersion version2
+
+    if v1.major < v2.major or v1.minor < v2.minor or v1.patch < v2.patch
+        return -1
+    else if v1.major > v2.major or v1.minor > v2.minor or v1.patch > v2.patch
+        return 1
+    else return 0
```

Comment:
@jacquarg this methods is wrong. You should add unit tests for it.

For example:

```coffee
v1 = [2, 0, 0]
v2 = [1, 2, 3]
compareVersions v1, v2  # it returns -1, instead of 1
```